From e133646d2e582d374a1fb0b17fb41c6a89aadd19 Mon Sep 17 00:00:00 2001
From: LuK1337 <priv.luk@gmail.com>
Date: Fri, 26 Feb 2016 14:12:55 +0100
Subject: [PATCH 065/171] Snap: Do not crash when cur-focus-scale is null

* Some cameras simply don't set this parameter.
* Also set cur-focus-diopter to 0 if it's null.

Change-Id: Ib1049012de9a58279560c14ed77f83f52c07fe30
---
 src/com/android/camera/PhotoModule.java | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/src/com/android/camera/PhotoModule.java b/src/com/android/camera/PhotoModule.java
index 78bb75ddf..c43449d9e 100644
--- a/src/com/android/camera/PhotoModule.java
+++ b/src/com/android/camera/PhotoModule.java
@@ -4416,7 +4416,14 @@ public class PhotoModule
             final int maxFocusPos = mParameters.getInt(CameraSettings.KEY_MAX_FOCUS_SCALE);
             //update mparameters to fetch latest focus position
             mParameters = mCameraDevice.getParameters();
-            final int CurFocusPos = mParameters.getInt(CameraSettings.KEY_MANUAL_FOCUS_SCALE);
+            int CurFocusPos = minFocusPos;
+
+            try {
+                 CurFocusPos = mParameters.getInt(CameraSettings.KEY_MANUAL_FOCUS_SCALE);
+            } catch (NumberFormatException e) {
+                 // Do nothing
+            }
+
             focusbar.setProgress(CurFocusPos);
             focusPositionText.setText("Current focus position is " + CurFocusPos);
 
@@ -4468,7 +4475,8 @@ public class PhotoModule
             //update mparameters to fetch latest focus position
             mParameters = mCameraDevice.getParameters();
             final String CurFocusPos = mParameters.get(CameraSettings.KEY_MANUAL_FOCUS_DIOPTER);
-            focusPositionText.setText("Current focus position is " + CurFocusPos);
+            focusPositionText.setText("Current focus position is " +
+                    (CurFocusPos != null ? CurFocusPos : minFocusStr));
             linear.addView(input);
             linear.addView(focusPositionText);
             alert.setView(linear);
-- 
2.17.1

